### ISSUE TITLE
`finalize_event` does not work on `HierarchicalMachine`

### ISSUE THREAD

**Issue by kunalbhagawati** [tyarkoni/transitions#216]
I have a state machine defined like so:
```
class CheckoutCoordinator(HierarchicalMachine):
    """
    Coordinator to handle the checkout flow.
    Implemented as a state machine that looks up its own current state and figures out what to do next.
    """

    def __init__(self):
        self.state_computer = CoordinatorStateComputer(self)
        super(CheckoutCoordinator, self).__init__(states=PROCESS_STATES,
                                                  transitions=TRANSITIONS,
                                                  after_state_change=['save_state',
                                                                      'attempt_to_release_checkout_area',
                                                                      'attempt_to_release_user'],
                                                  finalize_event=['attempt_to_release_user'],
                                                  send_event=True)
```

Now I assume `attempt_to_release_user` should always be called regardless of the state changing or not. But this is not happening.
Am i doing this wrong, and there is another way? I need a mechanism to execute something after the machine has done its work

**Comment by kunalbhagawati** [comment]
It seems `NestedEvent._trigger` is not calling `self.machine.finalize_event`. Can this be the issue?

**Comment by kunalbhagawati** [comment]
Sorry just saw the latest pull. Closing this.

**Comment by aleneum** [comment]
yeah, sounds like #211. It should have been fixed in `0.5.1`. If that does not work for you, feel free to reopen the issue.
